rfc/text/advanced/ap_50_protected_payload_mode.md

```diff
@@ -0,0 +1,284 @@
+## Protected Payload Mode
+
+In some situations, you may want to reduce the access the router has to the information users transmit, or
+payload data is presented in some specific format that can not be simply recognized by WAMP router serializer.
+
+These are example use cases:
+* Using WAMP alongside with gateways to other technologies like MQTT Brokers or AMQP Queues. So actual payload 
+  is for example mqtt message that should be simply delivered to wamp topic as is. 
+* Sensitive user data that should be delivered to target *Callee* without any possibility to unveil it during delivery.
+
+This needs can be covered with `Protected Payload Mode` feature. This feature allows:
+
+* Specifying additional attributes during `CALL`, `PUBLISH`, `EVENT`, `YIELD`, `RESULT` messages 
+  to signal *Router* to skip payload inspection.
+* Encrypting and decrypting payload with crypto algorithms.
+* Providing additional information about payload format and type.
+
+**Feature Announcement**
+
+Support for this advanced feature MUST be announced by *Callers* (`role := "caller"`), *Callees* (`role := "callee"`),
+*Dealers* (`role := "dealer"`), *Publishers* (`role := "publisher"`), *Subscribers* (`role := "subscriber"`) 
+and *Brokers* (`role := "broker"`) via
+
+{align="left"}
+HELLO.Details.roles.<role>.features.protected_payload_mode|bool := true
```

Comment:
so this doesn't announce the specific way it will be used in the client, e.g. it doesn't say

```
protected_payload_mode = "eth.xsalsa20poly1305.cbor"
```

or

```
protected_payload_mode = "mqtt.native"
```
or

```
protected_payload_mode = ["eth.xsalsa20poly1305.cbor", "eth.xsalsa20poly1305.flatbuffers"]
```